impl/c-qube/src/services/csv-adapter/csv-adapter.service.ts

```diff
@@ -584,26 +671,45 @@ export class CsvAdapterService {
         `TRUNCATE table spec."EventGrammar" CASCADE;`,
       );
 
+
       const dimensions: any[] = await this.prisma
         .$queryRaw`select 'drop table if exists "' || tablename || '" cascade;'
         from pg_tables where schemaname = 'dimensions';`;
+      
+        const datasets: any[] = await this.prisma
+        .$queryRaw`select 'drop table if exists "' || tablename || '" cascade;'
+        from pg_tables where schemaname = 'datasets';`;
+        
+        const totalIterations = dimensions.length + datasets.length ;
+
+        const p1 = new cliProgress.SingleBar({
```

Comment:
since this is being called at a number of places, abstract this out as a function, so that it can be used in the future scope as well.